pandas/io/excel/_base.py

```diff
@@ -895,14 +898,7 @@ def sheet_names(self):
 
     def close(self):
         """close io if necessary"""
-        if self.engine == "openpyxl":
-            # https://stackoverflow.com/questions/31416842/
-            #  openpyxl-does-not-close-excel-workbook-in-read-only-mode
-            wb = self.book
-            wb._archive.close()
-
-        if hasattr(self.io, "close"):
-            self.io.close()
```

Comment:
Is this intentionally removed? Wondering if there is any engine that needs this...

Reply:
This is the bug fix